src/Cli/dotnet/commands/dotnet-info/InfoCommand.cs

```diff
@@ -0,0 +1,197 @@
+
+namespace Microsoft.DotNet.Tools.Info
+{
+    public class InfoCommand
+    {
+        private readonly ParseResult _parseResult;
+
+        public InfoCommand(ParseResult parseResult)
+        {
+            _parseResult = parseResult;
+        }
+
+        public static int Run(ParseResult result)
+        {
+            result.HandleDebugSwitch();
+            var format = result.GetValue(InfoCommandParser.FormatOption);
+            if (format != InfoCommandParser.FormatOptions.json) {
+                PrintInfo();
+            } else  {
+                // To be implemented
+                PrintJsonInfo();
+            }
+            return 0;
+        }
+
+        public static void PrintVersion()
+        {
+            Reporter.Output.WriteLine(Product.Version);
+        }
+
+        public static void PrintInfo()
+        {
+            DotnetVersionFile versionFile = DotnetFiles.VersionFileObject;
+            var commitSha = versionFile.CommitSha ?? "N/A";
+            Reporter.Output.WriteLine($"{LocalizableStrings.DotNetSdkInfoLabel}");
+            Reporter.Output.WriteLine($" Version:           {Product.Version}");
+            Reporter.Output.WriteLine($" Commit:            {commitSha}");
+            Reporter.Output.WriteLine($" Workload version:  {WorkloadCommandParser.GetWorkloadsVersion()}");
+            Reporter.Output.WriteLine();
+            Reporter.Output.WriteLine($"{LocalizableStrings.DotNetRuntimeInfoLabel}");
+            Reporter.Output.WriteLine($" OS Name:     {RuntimeEnvironment.OperatingSystem}");
+            Reporter.Output.WriteLine($" OS Version:  {RuntimeEnvironment.OperatingSystemVersion}");
+            Reporter.Output.WriteLine($" OS Platform: {RuntimeEnvironment.OperatingSystemPlatform}");
+            Reporter.Output.WriteLine($" RID:         {GetDisplayRid(versionFile)}");
+            Reporter.Output.WriteLine($" Base Path:   {AppContext.BaseDirectory}");
+            PrintWorkloadsInfo();
+        }
+
+        private static void PrintWorkloadsInfo()
+        {
+            Reporter.Output.WriteLine();
+            Reporter.Output.WriteLine($"{LocalizableStrings.DotnetWorkloadInfoLabel}");
+            WorkloadCommandParser.ShowWorkloadsInfo();
+        }
+
+        private static string GetDisplayRid(DotnetVersionFile versionFile)
+        {
+            FrameworkDependencyFile fxDepsFile = new();
+
+            string currentRid = RuntimeInformation.RuntimeIdentifier;
+
+            // if the current RID isn't supported by the shared framework, display the RID the CLI was
+            // built with instead, so the user knows which RID they should put in their "runtimes" section.
+            return fxDepsFile.IsRuntimeSupported(currentRid) ?
+                currentRid :
+                versionFile.BuildRid;
+        }
+
+        public class RuntimeEnvironmentInfo
+        {
+            public string Name { get; set; }
```

Comment:
Please use [JsonPropertyName](https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/customize-properties?pivots=dotnet-8-0#customize-individual-property-names) to set the camelCase name style defined in baronfel's original issue.